src/CortexClient.cpp

```diff
@@ -0,0 +1,245 @@
+{
+  if (param.get_type() != rcl_interfaces::msg::ParameterType::PARAMETER_STRING) {
+    RCLCPP_ERROR(this->get_logger(), "Invalid parameter type for parameter %s",
+      param.get_name().c_str());
+    return false;
+  }
+
+  std::string temp_path = param.as_string();
+
+  if (getFileExtension(temp_path) != "json") {
+    RCLCPP_ERROR(this->get_logger(), "Invalid file format for parameter %s",
+      param.get_name().c_str());
+    return false;
+  }
+  capture_file_path_ = temp_path;
+  cortex_mock_ = CortexMock(capture_file_path_);
+  return true;
+}
+
+bool CortexClient::onRequestCommandChanged(const rclcpp::Parameter & param)
+{
+  if (param.get_type() != rcl_interfaces::msg::ParameterType::PARAMETER_STRING) {
+    RCLCPP_ERROR(this->get_logger(), "Invalid parameter type for parameter %s",
+      param.get_name().c_str());
+    return false;
+  }
+
+  std::string req_comm = param.as_string();
+  void * p_response = nullptr;
+  cortex_mock_.request(&req_comm[0], &p_response, nullptr);
+  if (req_comm == "PostGetPlayMode" || req_comm == "GetContextAnalogBitDepth" ||
+    req_comm == "GetUpAxis")
+  {
+    RCLCPP_INFO(get_logger(),
+      "Result of request " + req_comm + ": " + std::to_string(*static_cast<int *>(p_response)));
+  } else {  // Formatted this way, so that both cpplint and uncrustify can pass
+    if (req_comm == "GetContextFrameRate" ||
+      req_comm == "GetContextAnalogSampleRate" ||
+      req_comm == "GetConversionToMillimeters")
+    {
+      RCLCPP_INFO(get_logger(),
+        "Result of request " + req_comm + ": " + std::to_string(*static_cast<float *>(p_response)));
+    } else if (req_comm == "GetFrameOfData") {
+      sFrameOfData fod;
+      cortex_mock_.copyFrame(static_cast<sFrameOfData *>(p_response), &fod);
+      RCLCPP_INFO(get_logger(), "Frame " + std::to_string(fod.iFrame));
+      RCLCPP_INFO(get_logger(),
+        "Number of unidentified markers " + std::to_string(fod.nUnidentifiedMarkers));
+    }
+  }
+
+  return true;
+}
+
+void CortexClient::dataHandlerFunc_(sFrameOfData * fod)
+{
+  cortex_mock_.copyFrame(fod, &current_fod_);
+  RCLCPP_INFO(get_logger(), "Frame " + std::to_string(current_fod_.iFrame));
+  RCLCPP_INFO(get_logger(),
+    "Number of unidentified markers " + std::to_string(current_fod_.nUnidentifiedMarkers));
+}
+
+void CortexClient::errorMsgHandlerFunc_(int i_level, char * error_msg)
+{
+  switch (i_level) {
+    case 1:
+      RCLCPP_ERROR(get_logger(), static_cast<std::string>(error_msg));
+      break;
+    case 2:
+      RCLCPP_WARN(get_logger(), static_cast<std::string>(error_msg));
+      break;
+    case 3:
+      RCLCPP_INFO(get_logger(), static_cast<std::string>(error_msg));
+      break;
+    case 4:
+      RCLCPP_DEBUG(get_logger(), static_cast<std::string>(error_msg));
+      break;
+    default:
+      break;
+  }
+}
+
+}  // namespace ros2_cortex
+
+int main(int argc, char const * argv[])
+{
+  rclcpp::init(argc, argv);
+  rclcpp::executors::MultiThreadedExecutor executor;
```

Comment:
I don't see a reason to use a MultiThreadedExecutor. It is needed when multiple ROS callbacks have to be executed concurrently, but it is not the case here (well, there are neither subscribers, nor publishers at all yet :D ). A SingleThreadedExecutor should work fine.
```suggestion
  rclcpp::executors::SingleThreadedExecutor executor;
```